flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaConsumerTestBase.java

```diff
@@ -1238,7 +1238,7 @@ public void runProduceConsumeMultipleTopics(boolean useLegacySchema) throws Exce
         // create topics with content
         final List<String> topics = new ArrayList<>();
         for (int i = 0; i < numTopics; i++) {
-            final String topic = topicNamePrefix + i;
+            final String topic = topicNamePrefix + i + UUID.randomUUID();
```

Comment:
```suggestion
            final String topic = String.format("%s-%d-%s", topicNamePrefix, i, UUID.randomUUID());
```
sorry for being nitpicky about this specific one but I feel like here it's even more crucial that we add a separator since we provide this `i` identifier to each topic which would be kind of swallowed into the UUID.

Reply:
I noticed that we have also other occurrences where we're not adding the UUID to the topic (I didn't check that in my first pass). Is there a specific reason why we only select certain topics? Alternatively, we could add the randomization of topics to [KafkaTestBase.createTestTopic](https://github.com/apache/flink/blob/44f73c496ed1514ea453615b77bee0486b8998db/flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestBase.java#L215) and [KafkaTableTestBase](https://github.com/apache/flink/blob/ca9fea2459f7e066eac0fb59b382e58c41e0d702/flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/KafkaTableTestBase.java#L132). We would have to change the methods' signatures to return the randomized topic String (we should rename the method name in that case as well) which will then be used in the corresponding tests:
```
// old method signature:
public void createTestTopic(String topic, int numPartitions, int replicationFactor) {
// ...
// new method signature:
public String createRandomizedTestTopic(String topicPrefix, int numPartitions, int replicationFactor) {
```
WDYT?

Reply:
There were several places where random was not added so now added them manually.
Having a function which is not generic I'm not favoring. If we have `createRandomizedTestTopic` then we enforce devs to use randomness but sometimes one may need fixed names. That's the reason why I say let's have an end-to-end concept.

W/o too much consideration I would vote later on to have a `generateRandomTopicName(String prefix)` and `createTopic(String topic, int numPartitions, int replicationFactor)`. This way one can play lego based on needs.

There are some leftovers where no random added because these doesn't have restart so they either pass or fail.

Reply:
> sorry for being nitpicky about this specific one but I feel like here it's even more crucial that we add a separator since we provide this i identifier to each topic which would be kind of swallowed into the UUID.
I see, makes sense. You missed my initial comment, though. I shouldn't have added the second comment to this thread, I guess. Sorry for that one. But I think that the separator is important in the case above.

Reply:
Ah, made it 10pm w/ super tired. Added it.